src/main/java/org/broadinstitute/hellbender/utils/GATKProtectedMathUtils.java

```diff
@@ -157,13 +159,101 @@ public static double stdDev(final double ... values) {
         return Math.sqrt(new Variance().evaluate(values));
     }
 
+    /**
+     * Create a set number of points, linearly spaced, between a minimum and maximum.
+     *
+     * http://stackoverflow.com/questions/6208878/java-version-of-matlabs-linspace
+     *
+     * NaN are allowed, but will likely give useless results.
+     *
+     * @param min starting value
+     * @param max ending value
+     * @param points number of points, must be greater than -1
+     * @return Never {@code null}
+     */
+    public static double[] linspace(final double min, final double max, int points) {
+        ParamUtils.isPositiveOrZero(points, "Number of points must be >= 0");
+        if (points == 1) {
+            return new double[] {max};
+        }
+        double[] d = new double[points];
+        for (int i = 0; i < points; i++){
+            d[i] = min + i * (max - min) / (points - 1);
+        }
+        return d;
+    }
+
     // given a list of options and a function for calculating their probabilities (these must sum to 1 over the whole list)
     // randomly choose one option from the implied categorical distribution
     public static <E> E randomSelect(final List<E> choices, final Function<E, Double> probabilityFunction, final RandomGenerator rng) {
         final List<Pair<E, Double>> pmf = choices.stream()
                 .map(e -> new Pair<>(e, probabilityFunction.apply(e))).collect(Collectors.toList());
         return new EnumeratedDistribution<>(rng, pmf).sample();
     }
+
+    /**
+     *  Return an array with column sums in each entry
+     *
+     * @param matrix Never {@code null}
+     * @return Never {@code null}
+     */
+    public static double[] columnSum(final RealMatrix matrix) {
+        Utils.nonNull(matrix);
+
+        final Sum sumEvaluator = new Sum();
+        return IntStream.range(0, matrix.getColumnDimension())
+                .mapToDouble(c -> sumEvaluator.evaluate(matrix.getColumn(c))).toArray();
+    }
+
+    /**
+     *  Return an array with column sums in each entry
+     *
+     * @param matrix Never {@code null}
+     * @return Never {@code null}
+     */
+    public static double[] rowSum(final RealMatrix matrix) {
+        Utils.nonNull(matrix);
+
+        final Sum sumEvaluator = new Sum();
+        return IntStream.range(0, matrix.getRowDimension())
+                .mapToDouble(c -> sumEvaluator.evaluate(matrix.getRow(c))).toArray();
+    }
+
+
+    /**
+     *  Return sum of array
+     *
+     * @param array Never {@code null}
+     * @return sum of array
+     */
+    public static double sum(final double[] array) {
```

Comment:
Not crucial.  Still have methods named ``sum`` that handle multidimensional arrays, which are not in GATK public.